File: backend/src/main/java/PolarityBasic.java
import com.aliasi.classify.Classification;
import com.aliasi.classify.DynamicLMClassifier;
import com.aliasi.util.AbstractExternalizable;
import com.aliasi.util.Files;

import java.io.*;

public class PolarityBasic {
    File mPolarityDir;
    String[] mCategories;
   static DynamicLMClassifier mClassifier ;

    PolarityBasic(String[] args) {
        System.out.println("\nBASIC POLARITY DEMO");
        this.mPolarityDir = new File("training", "txt_sentoken");
        System.out.println("\nData Directory=" + this.mPolarityDir);
        this.mCategories = this.mPolarityDir.list();
        int nGram = 8;
        boolean bounded = false;
        this.mClassifier = new DynamicLMClassifier(this.mCategories, nGram, bounded);
    }

    void run() throws ClassNotFoundException, IOException {
        this.train();
       this.evaluate("fady");
    }

    boolean isTrainingFile(File file) {
        return file.getName().charAt(2) != '9';
    }

    void train() throws IOException {
        int numTrainingCases = 0;
        int numTrainingChars = 0;
        System.out.println("\nTraining.");
        for (int i = 0; i < this.mCategories.length; ++i) {
            String category = this.mCategories[i];
            File file = new File(this.mPolarityDir, this.mCategories[i]);
            File[] trainFiles = file.listFiles();
            for (int j = 0; j < trainFiles.length; ++j) {
                File trainFile = trainFiles[j];
                if (!this.isTrainingFile(trainFile)) continue;
                ++numTrainingCases;
                String review = Files.readFromFile((File)trainFile);
                numTrainingChars += review.length();
                this.mClassifier.train(category, (CharSequence)review);
            }
        }
        System.out.println("  # Training Cases=" + numTrainingCases);
        System.out.println("  # Training Chars=" + numTrainingChars);
//        try {
//            OutputStream file = new FileOutputStream("polarityBasicModelSmall.ser");
//            OutputStream buffer = new BufferedOutputStream(file);
//            ObjectOutput output = new ObjectOutputStream(buffer);
//
//            mClassifier.compileTo(output);
//        }
//
//        catch(IOException ex){
//            System.out.println("Error during writing object" + ex.getMessage());
//            ex.printStackTrace();
//        }

//        AbstractExternalizable.compileTo(mClassifier, classifierFile);
    }

    public static String evaluate(String tweet) throws IOException {
//        System.out.println("\nEvaluating.");
//        int numTests = 0;
//        int numCorrect = 0;
//        for (int i = 0; i < this.mCategories.length; ++i) {
//            String category = this.mCategories[i];
//            File file = new File(this.mPolarityDir, this.mCategories[i]);
//            File[] trainFiles = file.listFiles();
//            for (int j = 0; j < trainFiles.length; ++j) {
//                File trainFile = trainFiles[j];
//                if (this.isTrainingFile(trainFile)) continue;
//                String review = Files.readFromFile((File)trainFile);
//                ++numTests;
//                Classification classification = this.mClassifier.classify((Object)review);
//                if (!classification.bestCategory().equals(category)) continue;
//                ++numCorrect;
//            }
//        }
//        System.out.println("  # Test Cases=" + numTests);
//        System.out.println("  # Correct=" + numCorrect);
//        System.out.println("  % Correct=" + (double)numCorrect / (double)numTests);
//        System.out.println("\n Evaluating");
//        String review =  "Dear people who write spoilers online for #GoT. I hate you. I hope whatever happened to that character happens to you.";
//        Classification classification = mClassifier.classify((Object)review);
//        System.out.println(classification.bestCategory());
//


            Classification classification1 = mClassifier.classify((Object)tweet);
            return classification1.bestCategory();


//        File file = new File("polarityBasicModel.ser");
//
//        try {
//            mClassifier = (DynamicLMClassifier) AbstractExternalizable.readObject(file);
//        } catch (ClassNotFoundException e) {
//            e.printStackTrace();
//        }
//        String review =  "Dear people who write spoilers online for #GoT. I hate you. I hope whatever happened to that character happens to you.";
//        Classification classification = this.mClassifier.classify((Object)review);
//        System.out.println(classification.bestCategory());
    }

//    public static void main(String[] args) {
//        try {
//            new PolarityBasic(args).run();
//        }
//        catch (Throwable t) {
//            System.out.println("Thrown: " + t);
//            t.printStackTrace(System.out);
//        }
//    }
}

File: backend/src/main/scala/utilities/PolarityBasic.scala
package utilities

import java.io.{IOException, File}
import java.util
import java.util.Properties

import com.aliasi.classify.DynamicLMClassifier
import com.aliasi.util.Files
import edu.stanford.nlp.ling.CoreAnnotations.SentencesAnnotation
import edu.stanford.nlp.pipeline.{Annotation, StanfordCoreNLP}
import edu.stanford.nlp.sentiment.SentimentCoreAnnotations.SentimentClass
import edu.stanford.nlp.util.CoreMap
import scala.collection.JavaConversions._

object PolarityBasic {
  val props = new Properties()

  props.setProperty("annotators", "tokenize, ssplit, pos, lemma, parse, sentiment");
  val pipeline: StanfordCoreNLP = new StanfordCoreNLP(props);


  var mPolarityDir: File  = null
  var mCategories: Array[String] = null
  var mClassifier: DynamicLMClassifier = null

    println("\nBASIC POLARITY DEMO")
    this.mPolarityDir = new File("/home/ahmed/training", "txt_sentoken")
    println("\nData Directory=" + this.mPolarityDir)
    this.mCategories = this.mPolarityDir.list
    val nGram: Int = 8
    val bounded: Boolean = false
    this.mClassifier = new DynamicLMClassifier(this.mCategories, nGram, bounded)

  @throws(classOf[ClassNotFoundException])
  @throws(classOf[IOException])
  val run =  {
    this.train
  }

  def isTrainingFile(file: File): Boolean = {
    return file.getName.charAt(2) != '9'
  }

  @throws(classOf[IOException])
  def train {
    var numTrainingCases: Int = 0
    var numTrainingChars: Int = 0
    println("\nTraining.")

      var i: Int = 0
      while (i < this.mCategories.length) {

          val category: String = this.mCategories(i)
          val file: File = new File(this.mPolarityDir, this.mCategories(i))
          val trainFiles: Array[File] = file.listFiles()

            var j: Int = 0
            while (j < trainFiles.length) {
                val trainFile: File = trainFiles(j)
                if (!this.isTrainingFile(trainFile)){

                }
                else {
                  numTrainingCases += 1

                  val review: String = Files.readFromFile(trainFile.asInstanceOf[File])
                  numTrainingChars += review.length
                  this.mClassifier.train(category, review.asInstanceOf[CharSequence])
                }
              j += 1;
            }
          i += 1;
      }

    println("  # Training Cases=" + numTrainingCases)
    println("  # Training Chars=" + numTrainingChars)
  }

  @throws(classOf[IOException])
  def evaluate(tweet: String): String = {
    var sen:String = "good"
    val annotation: Annotation = pipeline.process(tweet);
    val sentences: util.List[CoreMap] = annotation.get(classOf[SentencesAnnotation])

    for (sentence :CoreMap <- sentences) {
      val sentiment: String = sentence.get(classOf[SentimentClass])

      if (sentiment == "Neutral") sen = sentiment
      else sen = mClassifier.classify(tweet.asInstanceOf[AnyRef]).bestCategory
    }
    sen
  }

}